cli/index.ts

```diff
@@ -1,45 +1,38 @@
 import { __ } from '@wordpress/i18n';
-import { Command, Option } from 'commander';
 import { StatsGroup, StatsMetric } from 'common/types/stats';
-import { registerCommand as registerPreviewCreateCommand } from 'cli/commands/preview/create';
-import { registerCommand as registerPreviewDeleteCommand } from 'cli/commands/preview/delete';
-import { registerCommand as registerPreviewListCommand } from 'cli/commands/preview/list';
-import { registerCommand as registerPreviewUpdateCommand } from 'cli/commands/preview/update';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { registerPreviewCommands } from 'cli/commands/preview';
 import { loadTranslations } from 'cli/lib/i18n';
 import { bumpAggregatedUniqueStat } from 'cli/lib/stats';
 import { version } from 'cli/package.json';
+import { OutputFormat, StudioArgv } from 'cli/types';
 
 async function main() {
-	await loadTranslations();
+	const locale = await loadTranslations();
 
-	const program = new Command();
-
-	program
-		.name( 'studio' )
-		.description( __( 'Studio by WordPress.com CLI' ) )
+	const studioArgv: StudioArgv = yargs( hideBin( process.argv ) )
+		.scriptName( 'studio' )
+		.usage( __( 'Studio by WordPress.com CLI' ) )
+		.locale( locale )
 		.version( version )
-		.hook( 'preAction', () => {
-			bumpAggregatedUniqueStat( StatsGroup.STUDIO_CLI_USAGE_UNIQUE, StatsMetric.SUCCESS, 'weekly' );
-		} )
-		.addOption(
-			new Option( '--output-format [format]', __( 'Specify a non-standard output format' ) )
-				.argParser( ( value: string ) => {
-					if ( value !== 'json' ) {
-						throw new Error( __( 'The only custom output format supported is "json"' ) );
-					}
-					return value;
-				} )
-				.hideHelp()
-		);
-
-	const previewCommand = program.command( 'preview' ).description( __( 'Manage preview sites' ) );
+		.middleware( async () =>
+			bumpAggregatedUniqueStat( StatsGroup.STUDIO_CLI_USAGE_UNIQUE, StatsMetric.SUCCESS, 'weekly' )
+		)
+		.option( 'output-format', {
+			type: 'string',
+			hidden: true,
+			coerce: ( value: string ): OutputFormat => {
+				if ( value !== 'json' ) {
+					throw new Error( __( 'The only custom output format supported is "json"' ) );
+				}
+				return value;
+			},
+		} );
 
-	registerPreviewCreateCommand( program );
-	registerPreviewListCommand( previewCommand );
-	registerPreviewDeleteCommand( previewCommand );
-	registerPreviewUpdateCommand( previewCommand );
+	registerPreviewCommands( studioArgv ).strict().help();
 
-	program.parse( process.argv );
+	await studioArgv.argv;
```

Comment:
```ts
		} )
		.demandCommand( 1, __( 'You need to specify a command' ) )
		.strict();

	registerPreviewCommands( studioArgv );
```
When I run the CLI script without any arguments, the help page isn't printed right now. Adding a `demandCommand` call fixes that.